aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,486 @@
+
+The Validation Summary Tag Helper
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The `Validation Summary Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/ValidationSummaryTagHelper/index.html>`__  is used to display a summary of validation messages. The `asp-validation-summary`` attribute value can be any of the following:
+
++-------------------------------+--------------------------------+
+|asp-validation-summary         |  Validation messages displayed |  
++===============================+================================+
+|ValidationSummary.All          | Property and model level       |
++-------------------------------+--------------------------------+  
+|ValidationSummary.ModelOnly    | Model                          |
++-------------------------------+--------------------------------+  
+|ValidationSummary.None         | None                           |
++-------------------------------+--------------------------------+  
+
+If the form has validation errors, the browser reports the validation errors and the form data is not posted to the server.
+
+Example
+^^^^^^^^^
+
+In the following example, the data model is decorated with ``DataAnnotations`` attributes, which generates validation error messages on the ``<input>`` element.  When a validation error occurs, the Validation Tag Helper displays the error message:
+
+.. literalinclude::  forms/sample/final/ViewModels/RegisterViewModel.cs
+  :language: c#
+
+..  literalinclude::  forms/sample/final/Views/Demo/RegisterValidation.cshtml
+  :language: HTML
+  :emphasize-lines: 4,6,8
+  :lines: 1-10
+
+The generated HTML:
+
+.. code-block:: HTML
+  :emphasize-lines: 2,3,8,9,12,13
+  :linenos:
+  
+  <form action="/DemoReg/Register" method="post">
+    <div class="validation-summary-valid" data-valmsg-summary="true">
+    <ul><li style="display:none"></li></ul></div>
+    Email:  <input name="Email" id="Email" type="email" value="" 
+     data-val-required="The Email field is required." 
+     data-val-email="The Email field is not a valid e-mail address." 
+     data-val="true"> <br>
+    <span class="field-validation-valid" data-valmsg-replace="true" 
+     data-valmsg-for="Email"></span><br>
+    Password: <input name="Password" id="Password" type="password" 
+     data-val-required="The Password field is required." data-val="true"><br>
+    <span class="field-validation-valid" data-valmsg-replace="true" 
+     data-valmsg-for="Password"></span><br>
+    <button type="submit">Register</button>
+    <input name="__RequestVerificationToken" type="hidden" value="<removed for brevity>" />
+  </form>
+
+The Select Tag Helper
+-------------------------
+
+.. question One odd thing I noticed about this is that "asp-for" doesn't require the Model. prefix, but "asp-items" does. If anybody knows why this is, I'd love to hear it.  http://www.exceptionnotfound.net/tag-helpers-in-asp-net-5-an-overview/#selecttaghelper 
```

Comment:
`asp-for="@Model.Property"` is also supported but is equivalent to `m => Model.Property` i.e. it's still part of a lambda expression. `asp-items="Model.Property"` is instead part of an assignment: `tagHelper.Items = Model.Property`. Put another way `asp-for` is special and there is nothing else like that attribute in the MVC tag helpers.